.github/workflows/build-and-test.yml

```diff
@@ -5,42 +5,92 @@ on:
     branches: [dev]
   pull_request:
     branches: [dev]
-
 jobs:
-  run-goreleaser:
-    name: "Build All Binaries and Frameworks"
-    runs-on: ubuntu-latest
+  run-bulid:
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+        go-version: [1.18, 1.19]
+        ignite-version: [v0.22.1, v0.23.0, v0.24.0]
+    name: "Build/Test ${{ matrix.os }}; Go ${{ matrix.go-version }}; Ignite ${{ matrix.ignite-version }}"
+    runs-on: ${{ matrix.os }}
     steps:
       - name: Checkout
         uses: actions/checkout@v3
         with:
           fetch-depth: 0
 
-      # - name: Configure Homebrew w/ cache
-      #   uses: actions/cache@v3
-      #   with:
-      #     path: |
-      #       ~/Library/Caches/Homebrew/foo--*
-      #       ~/Library/Caches/Homebrew/downloads/*--foo-*
-      #     key: brew-${{ hashFiles('.github/brew-formulae') }}
-      #     restore-keys: brew-
-      # - name: Install Homebrew dependencies
-      #   run: |
-      #     brew install FiloSottile/musl-cross/musl-cross
-      #     brew install mingw-w64
-
-      - name: Run coverage
+      - uses: actions/setup-go@v3
+        with:
+          go-version: ~${{ matrix.go-version }}
+
+      - name: Install Ignite
+        run: |
+          sudo curl https://get.ignite.com/cli@${{matrix.ignite-version}}! | sudo bash
+
+      - name: Run build against Release target for macos and linux
+        continue-on-error: false
+        run: |
+          ignite chain build --release -t linux:amd64 -t darwin:amd64
+
+      - name: Run tests and evaluate coverage
         continue-on-error: true
         run: go test -race -coverprofile=coverage.out -covermode=atomic ./...
 
-      - name: Prepare Release Variables
-        id: vars
-        uses: ignite-hq/cli/actions/release/vars@develop
+      - name: Upload coverage to Codecov
+        uses: codecov/codecov-action@v3
+
+
+  bind-them-all:
+    strategy:
+      matrix:
+        platform: [ios, web, android]
+        build-on: [ubuntu-latest, macos-latest]
+        go-version: [1.17, 1.18, 1.19]
+        exclude:
+        - build-on: ubuntu-latest
+          platform: ios
+        - build-on: macos-latest
+          platform: android
+      fail-fast: false
+    name: "Bindings: ${{ matrix.platform }}; ${{ matrix.build-on }}; Go ${{ matrix.go-version }}"
+    runs-on: ${{ matrix.build-on }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Force NDK version
+        if : matrix.platform == 'android'
+        run: |
+          echo "y" | sudo ${ANDROID_HOME}/tools/bin/sdkmanager --install "ndk;23.1.7779620" | grep -v = || true
```

Comment:
should `ndk;23.1.xxxxx` be moved to an env var?

Reply:
adding it to the matrix..... you know not what you've done